pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -306,7 +344,7 @@ func (h *handler) OnRemove(key string, obj runtime.Object) (runtime.Object, erro
 	}
 
 	if _, nodeDrainExcluded := machine.Annotations[capi.ExcludeNodeDrainingAnnotation]; !removed ||
-		(!nodeDrainExcluded && drainingSucceededCondition.IsFalse(machine) && drainingSucceededCondition.GetReason(machine) != capi.DrainingFailedReason) {
+		(!nodeDrainExcluded && !drainingSucceededCondition.IsTrue(machine) && drainingSucceededCondition.GetReason(machine) != capi.DrainingFailedReason) {
```

Comment:
I was running into the case where the draining condition had not been set. In which case, `IsFalse` would return `false`, the if-check would fail, and the infrastructure would be deleted.

This would cause issues with the CAPI machine controller trying to drain a node that no longer exists.

With this change, that piece of the `&&` only fails when `IsTrue` returns `true` which is the desired outcome.